### PR TITLE
 handwired dactyl with 2 pro micros added 

### PR DESCRIPTION
**Pull request by zwnk**
handwired dactyl with 2 pro micros added 

## Description
The original Dactyl uses a teensy mc. This version uses 2 pro micro like the Dactyl Manuform already does.

## Types of changes

- [ ] Core
- [ ] Bugfix
- [ ] New Feature
- [ ] Enhancement/Optimization
- [x] Keyboard (addition or update)
- [x] Keymap/Layout/Userspace (addition or update)
- [x] Documentation

### CODE REVIEW

File: keyboards/handwired/dactyl_promicro/keymaps/default/keymap.c [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+extern keymap_config_t keymap_config;
+
+#define _QWERTY 0
+#define _LOWER 1
+#define _RAISE 2
+
+#define RAISE MO(_RAISE)
+#define LOWER MO(_LOWER)
+
+#define _______ KC_TRNS
```
Comment (drashna) [inline keyboards/handwired/dactyl_promicro/keymaps/default/keymap.c]: These defines aren't necessary, as these are actually defined in the core.  You can safely remove them. 

https://github.com/qmk/qmk_firmware/blob/a0309db983150fd0197eb620a6ba552d90d29f93/quantum/quantum_keycodes.h#L30-L31


```suggestion
```

File: keyboards/handwired/dactyl_promicro/keymaps/impstyle/keymap.c [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+extern keymap_config_t keymap_config;
+
+#define _QWERTY 0
+#define _LOWER 1
+#define _RAISE 2
+
+#define RAISE MO(_RAISE)
+#define LOWER MO(_LOWER)
+
+#define _______ KC_TRNS
```
Comment (drashna) [inline keyboards/handwired/dactyl_promicro/keymaps/impstyle/keymap.c]: These defines aren't necessary, as these are actually defined in the core.  You can safely remove them. 

https://github.com/qmk/qmk_firmware/blob/a0309db983150fd0197eb620a6ba552d90d29f93/quantum/quantum_keycodes.h#L30-L31


```suggestion
```

File: keyboards/handwired/dactyl_promicro/readme.md [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+-----------------------------
+
+Do update the `F_CPU` parameter in `rules.mk` to `8000000` which reflects
+the frequency on the 3.3V board.
+
+Also, if the slave board is producing weird characters in certain columns,
+update the following line in `matrix.c` to the following:
+
+```
+// _delay_us(30);  // without this wait read unstable value.
```
Comment (drashna) [inline keyboards/handwired/dactyl_promicro/readme.md]:
```suggestion
// wait_us(30);  // without this wait read unstable value.
```

File: keyboards/handwired/dactyl_promicro/readme.md [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+_delay_us(300);  // without this wait read unstable value.
```
Comment (drashna) [inline keyboards/handwired/dactyl_promicro/readme.md]:
```suggestion
wait_us(300);  // without this wait read unstable value.
```

File: keyboards/handwired/dactyl_promicro/readme.md [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+EEPROM for the left and right halves.
+
+The EEPROM is used to store whether the
+half is left handed or right handed. This makes it so that the same firmware
+file will run on both hands instead of having to flash left and right handed
+versions of the firmware to each half. To flash the EEPROM file for the left
+half run:
+```
+avrdude -p atmega32u4 -P $(COM_PORT) -c avr109 -U eeprom:w:"./quantum/split_common/eeprom-lefthand.eep"
+// or the equivalent in dfu-programmer
```
Comment (drashna) [inline keyboards/handwired/dactyl_promicro/readme.md]: which would be:
```
make handwired/dactyl:default:dfu-split-left
make handwired/dactyl:default:dfu-split-right
```

File: keyboards/handwired/dactyl_promicro/keymaps/default/keymap.c (hunk 1 above)
Comment (zwnk) [reply keyboards/handwired/dactyl_promicro/keymaps/default/keymap.c]: I saw that in another pull request after i opened mine :) sry

Comment (drashna) [reply keyboards/handwired/dactyl_promicro/keymaps/default/keymap.c]: No worries!

File: keyboards/handwired/dactyl_promicro/readme.md (hunk 4 above)
Comment (drashna) [reply keyboards/handwired/dactyl_promicro/readme.md]: Speficially, "wait" is hardware agnostic, while delay is AVR only.

File: keyboards/handwired/dactyl_promicro/readme.md (hunk 5 above)
Comment (drashna) [reply keyboards/handwired/dactyl_promicro/readme.md]: .... I need to figure out how to do this for avrdude....

Comment (zwnk) [reply keyboards/handwired/dactyl_promicro/readme.md]: to be honest, it was late when i copied the dactyl manuform readme and adapted some stuff.
i tried that once with the manuform but went back to to master/slave mode due to missing features.
in my opinion we can delete that procedure from the readme.

Comment (drashna) [reply keyboards/handwired/dactyl_promicro/readme.md]: no worries :)

Comment (zwnk) [reply keyboards/handwired/dactyl_promicro/readme.md]: i removed the whole part and did some minor changes.

File: keyboards/handwired/dactyl_promicro/dactyl_promicro.h [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+#endif
+
+
+#define LAYOUT_6x6(\
+	L00, L01, L02, L03, L04, L05,                    R00, R01, R02, R03, R04, R05, \
+	L10, L11, L12, L13, L14, L15,                    R10, R11, R12, R13, R14, R15, \
+	L20, L21, L22, L23, L24, L25,                    R20, R21, R22, R23, R24, R25, \
+	L30, L31, L32, L33, L34, L35,                    R30, R31, R32, R33, R34, R35, \
+	L40, L41, L42, L43, L44, L45,                    R40, R41, R42, R43, R44, R45, \
+		 L51, L52, L53, L54, L55,                    R50, R51, R52, R53, R54      \
```
Comment (drashna) [inline keyboards/handwired/dactyl_promicro/dactyl_promicro.h]: Is this accurate? 

Wouldn't it be more like: 
```c
                            L55,L56,   R50,R51,                          \
                                L54,   R52,                              \
                        L53,L52,L51,   R55,R54,R53                      \
```

Comment (zwnk) [reply keyboards/handwired/dactyl_promicro/dactyl_promicro.h]: well, the matrix shows how it is wired not how it is physically placed on the board.
can change that to reflect the key placement on the board if wanted.

Comment (pelrun) [reply keyboards/handwired/dactyl_promicro/dactyl_promicro.h]: The LAYOUT macros are specifically intended to be a physical layout-to-matrix transform. Not actually using the real physical layout (or an approximation) is rather missing the point.

Comment (zwnk) [reply keyboards/handwired/dactyl_promicro/dactyl_promicro.h]: no problem, will change that ;)

**Comment by drashna** [comment]
Awesome, thanks!

**Pull request merged**